src/OmniSharp.Roslyn/BufferManager.cs

```diff
@@ -64,7 +66,20 @@ public async Task UpdateBufferAsync(Request request)
 
                     foreach (var documentId in documentIds)
                     {
-                        solution = solution.WithDocumentText(documentId, sourceText);
+                        var document = solution.GetDocument(documentId);
+
+                        // This can happen when the client sends a create request for a file that we created on the server,
+                        // such as RunCodeAction. Unfortunately, previous attempts to have this fully controlled by the vscode
+                        // client (such that it sent both create event and then updated existing text) wasn't successful:
+                        // vscode seems to always trigger an update buffer event before triggering the create event.
+                        if ((await document.GetTextAsync()).Length > 0 && isCreate && string.IsNullOrEmpty(buffer))
```

Comment:
it would be slightly more efficient to check `await document.GetTextAsync()).Length` last, because the other two conditions may short circuit and there will be no need to fetch the document text anymore